include/nvexec/stream_context.cuh

```diff
@@ -75,6 +89,94 @@ namespace nvexec {
     template <sender Sender>
     using ensure_started_th = __t<ensure_started_sender_t<__id<Sender>>>;
 
+    struct stream_domain {
+      
+      // start of debugging utility
+      template <typename T>
+      void print_type_name() const {
+        stdexec::print(std::declval<stdexec::__detail::__name_of<T>>());
+      }
+
+      template <typename... Types>
+      struct type_list { };
+
+      template <std::size_t Index, typename... Types>
+      using nth_type_of = std::tuple_element_t<Index, std::tuple<Types...>>;
+
+      void for_each_in_pack() const {
+      }
+
+      template <typename First, typename... Rest>
+      void for_each_in_pack() const {
+        print_type_name<First>();
+        for_each_in_pack<Rest...>();
+      }
+      // end of debugging utility 
+
+      // not sure if reconstitute should exist here in the domain
+      template <class Tag>
+      auto reconstitute() const;
+
+      template <>
+      auto reconstitute<reduce_t>() const {
+        return [&](auto data, auto sender) {
+          auto initT = stdexec::__nth_member<0>()(data);
+          auto fun = stdexec::__nth_member<1>()(data);
+          auto transformed = reduce_sender_t<decltype(sender), decltype(initT), decltype(fun)>(
+            {}, sender, initT, fun);
+          return transformed;
+        };
```

Comment:
Some thoughts:

1) Does the fact that hte compiler no longer complains about hte errors related to `__on::receiver_ref` not having a call operator thus indicate that the tree has been fully transformed, children and alike? I was going off of this assumption. 

transform_sender will rely on this specialization of reconstitute for every sender alongside their children. Now one caveat is... Am I not transforming the entire tree? If that's the case, wouldn't there be an error similar to the 1st point? The fact htat it's looking for an `apply` function, does that not indicate it has transformed the tree properly?